xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp

```diff
@@ -317,58 +317,96 @@ void CDVDSubtitlesLibass::ApplyStyle(style subStyle, renderOpts opts)
       // Set SecondaryColour may be used to prevent an onscreen collision
       style->SecondaryColour = subColor;
 
-      // Compute the outline color
-      UTILS::Color outlineColor = ConvColor(subStyle.outlineColor);
-
       // Configure the effects
+      double lineSpacing = 0.0;
       if (subStyle.borderStyle == BorderStyle::OUTLINE)
       {
-        style->BorderStyle = BORDER_STYLE_OUTLINE;
-        style->OutlineColour = outlineColor;
-        style->BackColour = ConvColor(UTILS::COLOR::BLACK); // Shadow color
-        style->Outline = 1 * scale;
-        style->Shadow = 0;
+        style->BorderStyle = ASS_BORDER_STYLE_OUTLINE;
+        style->Outline = (10.00 / 100 * subStyle.fontBorderSize) * scale;
+        style->OutlineColour = ConvColor(subStyle.fontBorderColor, subStyle.fontOpacity);
+        style->BackColour =
+            ConvColor(subStyle.shadowColor, subStyle.shadowOpacity); // Set the shadow color
+        style->Shadow = (10.00 / 100 * subStyle.shadowSize) * scale; // Set the shadow size
```

Comment:
Since we are implementing shadows, although it might be considered a feature request, blur is something modern streaming applications (like netflix) use by default. This is part of the ASS_Style type: https://github.com/libass/libass/blob/master/libass/ass_types.h#L79

Reply:
i have add it